packages/core/src/rules/oas3/component-name-unique.ts

```diff
@@ -46,32 +49,47 @@ export const ComponentNameUnique: Oas3Rule | Oas2Rule = (options) => {
           const resolvedRef = resolve(ref);
           if (!resolvedRef.location) return;
 
-          addComponentFromAbsoluteLocation(
-            typeName,
-            resolvedRef.location.absolutePointer.toString()
-          );
+          addComponentFromAbsoluteLocation(typeName, resolvedRef.location);
         }
       },
     },
     Root: {
       leave(root: Oas3Definition, ctx: UserContext) {
         components.forEach((value, key, _) => {
-          if (value.size > 1) {
+          if (value.absolutePointers.size > 1) {
             const component = getComponentFromKey(key);
             const optionComponentName = getOptionComponentNameForTypeName(component.typeName);
-            const definitions = Array.from(value)
-              .map((v) => `- ${v}`)
-              .join('\n');
-
+            const definitions = Array.from(value.absolutePointers).join(', ');
             const problem: Problem = {
-              message: `Component '${optionComponentName}/${component.componentName}' is not unique. It is defined at:\n${definitions}`,
+              message: `Component '${optionComponentName}/${component.componentName}' is not unique. It is defined at: ${definitions}`,
+            };
+            problem.location = {
+              ...value.locations.sort((a, b) =>
+                a.absolutePointer.localeCompare(b.absolutePointer)
+              )[0],
             };
 
             const componentSeverity = optionComponentName ? options[optionComponentName] : null;
             if (componentSeverity) {
               problem.forceSeverity = componentSeverity;
             }
             ctx.report(problem);
+            // value.locations.forEach((location) => {
```

Comment:
Please remove these commented lines if they are not needed

Reply:
Oh my, totally forgot to remove them 🙈 . Removed them now.